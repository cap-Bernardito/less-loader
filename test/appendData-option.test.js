import {
  compile,
  getCodeFromBundle,
  getCompiler,
  getErrors,
  getWarnings,
} from './helpers';

jest.setTimeout(30000);

describe('appendData option', () => {
  it('should work append data as function', async () => {
    const testId = './append-data.less';
    const compiler = getCompiler(testId, {
      appendData() {
        return `@color: coral;`;
      },
    });
    const stats = await compile(compiler);
    const codeFromBundle = getCodeFromBundle(stats, compiler);

    expect(codeFromBundle.css).toMatchSnapshot('css');
    expect(getWarnings(stats)).toMatchSnapshot('warnings');
    expect(getErrors(stats)).toMatchSnapshot('errors');
  });

  it('should work append data as string', async () => {
    const testId = './append-data.less';
    const compiler = getCompiler(testId, {
      appendData: `@color: coral;`,
    });
    const stats = await compile(compiler);
    const codeFromBundle = getCodeFromBundle(stats, compiler);

    expect(codeFromBundle.css).toMatchSnapshot('css');
    expect(getWarnings(stats)).toMatchSnapshot('warnings');
    expect(getErrors(stats)).toMatchSnapshot('errors');
  });
});
