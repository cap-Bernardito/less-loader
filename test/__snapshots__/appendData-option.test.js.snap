// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`appendData option should work append data as function: css 1`] = `
".background {
  color: coral;
}
"
`;

exports[`appendData option should work append data as function: errors 1`] = `Array []`;

exports[`appendData option should work append data as function: warnings 1`] = `Array []`;

exports[`appendData option should work append data as string: css 1`] = `
".background {
  color: coral;
}
"
`;

exports[`appendData option should work append data as string: errors 1`] = `Array []`;

exports[`appendData option should work append data as string: warnings 1`] = `Array []`;
